### PR TITLE
Fix --replace behavior regression

### LINKED ISSUE #13001: `--replace` throws error `error: update is marked constrained, but is not constrained to an update or a replace`
### What happened?

A user in community slack messaged that they were running into an error when using `--replace`

> Hi, I'm getting the following error when running pulumi with a --replace:
```
    error: update is marked constrained, but is not constrained to an update or a replace
```

This is likely due to the changes in step_generator.go

[community slack](https://pulumi-community.slack.com/archives/C84L4E3N1/p1684770323349649)

### Expected Behavior

Not run into an error.

### Steps to reproduce

.

### Output of `pulumi about`

v3.68.0

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

### PR DESCRIPTION
**Pull request by dixler**
# Description

`--replace` is not considered a targeted update. `--target-replace` works by adding the urn to the ReplaceTarget and also the UpdateTarget list. This commit removes a function that conflates the two and a test that incorrectly verifies this incorrect behavior.

Fixes #13001 

TODO:
- [ ] squash commits

## Checklist

- [x] I have run `make tidy` to update any new dependencies
- [x] I have run `make lint` to verify my code passes the lint check
  - [x] I have formatted my code using `gofumpt`

- [x] I have added tests that prove my fix is effective or that my feature works

- [x] I have run `make changelog` and committed the `changelog/pending/<file>` documenting my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Cloud API version

### CODE REVIEW

File: pkg/resource/deploy/step_generator.go [hunk 1]
```diff
@@ -307,7 +303,7 @@ func (sg *stepGenerator) GenerateSteps(event RegisterResourceEvent) ([]Step, res
 		}
 	}
 
-	if !sg.isTargetedUpdate() {
+	if !sg.updateTargetsOpt.IsConstrained() {
 		return steps, nil
 	}
```
Comment (dixler) [inline pkg/resource/deploy/step_generator.go:308]: This used to early exit for updates where `--replace` was specified...

This is a behavior change. Consider the repercussions.

Comment (dixler) [reply pkg/resource/deploy/step_generator.go]: I've removed the behavior change to avoid potential regressions.

**Review by Frassle — APPROVED** [review]
Seems right

**Comment by dixler** [comment]
bors merge

**Pull request merged**